### DISCUSSION TITLE
Oracle Cloud Deployment

### DISCUSSION THREAD

**Discussion by aalikadic** [fastapi/fastapi#2703] [Questions]
Hello everyone,

I am developing a ML model using FastAPI that has to be served on Oracle Cloud. I've been thru Google and Stack Overflow, but failed to find example guides. Does anyone have a proper guide/example on how to deploy FastAPI on Oracle Cloud other than using docker?

**Comment by haiderasad** [comment]
hi it should be 
1)first things first create the fastapi app in python 
2)wrap this app in a docker image with base image as cuda image(if you want to run ML on GPU)
3)make sure your remote machine has the same cuda and cudnn versions as on the docker image
4)through any docker registry push the created image to that registry and then from remote machine pull that image
5)good practice would be to use gunicorn with nginx to give your api proper routing or you can simple use uvicorn only 

refer to these sites for in detail instructions:
https://www.vultr.com/docs/how-to-deploy-fastapi-applications-with-gunicorn-and-nginx-on-ubuntu-20-04/

https://medium.com/@mingc.me/deploying-pytorch-model-to-production-with-fastapi-in-cuda-supported-docker-c161cca68bb8

https://towardsdatascience.com/containerizing-huggingface-transformers-for-gpu-inference-with-docker-and-fastapi-on-aws-d4a83edede2f
